components/Forms/form.tsx

```diff
@@ -3,7 +3,7 @@ import UpdateUserForm from "./updateUserForm";
 import { useSelector } from "react-redux";
 import { useReducer } from "react";
 
-const formReducer = (state:any, event:any) => {
+const formReducer = (state: any, event: any) => {
```

Comment:
Try as much as you can to avoid `any` typing. This defeats the purpose of using TypeScript. I know you are just stating but in this case, if you are passing state and a HTML event... I bet you can find the right types ;)

Reply:
Looked for this for a while, I'm not finding any type except any for the state.

made :
```suggestion
interface FormReducerProps {
  state: any;
  event: React.ChangeEvent<HTMLInputElement>
}
```
My state is getting spread, so type can only be a Function, Object or Array,  I am right ?

Reply:
The state that is being passed here has to be defined somewhere aka your context provider.